org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/desktop/DashboardController.java

```diff
@@ -1021,7 +1035,50 @@ else if (DisplayType.isDate(iPara.getDisplayType()))
 			}
 		}				
 	}
-	
+
+	private static final String displayColumns = "Name,Value,DocumentNo";
+	private String getDisplay(MPInstance i, MPInstancePara ip, Object id) {
+		try {
+			MProcessPara pp = MProcess.get(i.getAD_Process_ID()).getParameter(ip.getParameterName());
+			MTable table = null;
+			String column = "";
+
+			if (pp != null) {
+				if ((pp.getAD_Reference_ID() == DisplayType.TableDir || pp.getAD_Reference_ID() == DisplayType.Search) && pp.getAD_Reference_Value_ID() == 0)
+					table = MTable.get(Env.getCtx(), pp.getColumnName().replace("_ID", ""));
+
+				if (pp.getAD_Reference_Value_ID() > 0) {
+					MRefTable rt = MRefTable.get(pp.getAD_Reference_Value_ID());
+					if (table == null)
+						table = MTable.get(rt.getAD_Table_ID());
+					column = MColumn.getColumnName(Env.getCtx(), rt.getAD_Display());
+				}
+
+				if (table != null && table.isSingleKey() && table.getKeyColumns().length == 1) { // just to be sure
+
+					if (Util.isEmpty(column) || column.endsWith("_ID")) {
+						for (String columnName : displayColumns.split(",")) {
+							if (table.getColumnIndex(columnName) >= 0) {
+								column = columnName;
+								break;
+							}
+						}
+					}
+
+					StringBuilder sql = new StringBuilder("SELECT ").append(column).append(" FROM ").append(table.getTableName())
+							.append(" WHERE ").append(table.getKeyColumns()[0]).append(" = ?");
+
+					return DB.getSQLValueStringEx(null, sql.toString(), id);
+				}
+			}
+		}
+		catch (Exception e) {
+			logger.log(Level.WARNING, "Failed to retrieve data to display for embedded report " + MProcess.get(i.getAD_Process_ID()).getName() + " : " + ip.getParameterName(), e);
+		}
+
+		return id.toString();
```

Comment:
Instead of coding the display column stuff here again, I think we should use the getlookupinfo methods in mlookupfactory and use the MLookupInfo.QueryDirect to get the display value. This reduce the duplicate of code and at the same give better consistency with other part of the application.

Reply:
Something like that ?